apis/pipelines/v1alpha5/test_generators.go

```diff
@@ -32,6 +35,53 @@ func RandomPipelineSpec() PipelineSpec {
 	}
 }
 
+func RandomProvider() *Provider {
+	return &Provider{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RandomLowercaseString(),
+			Namespace: "default",
+		},
+		Spec:   RandomProviderSpec(),
+		Status: RandomProviderStatus(),
+	}
+}
+
+func RandomProviderSpec() ProviderSpec {
+	randomParameters := make(map[string]*apiextensionsv1.JSON)
+	for key, _ := range RandomMap() {
+		randomParameters[key] = &apiextensionsv1.JSON{Raw: []byte(`{"key1": "value1", "key2": 1234}`)}
```

Comment:
could randomise the keys and values?

Reply:
this might overcomplicate the spec, its just a placeholder to check the params are working. Due to the JSON type it gets messy trying to template it